drivers/hmis/app/models/hmis/hud/project.rb

```diff
@@ -57,7 +57,15 @@ class Hmis::Hud::Project < Hmis::Hud::Base
 
   has_many :services, through: :enrollments_including_wip
   has_many :custom_services, through: :enrollments_including_wip
-  has_many :hmis_services, through: :enrollments_including_wip
+
+  # FIXME: joining services through client_projects confounds postgres. On larger projects, the query might take many
+  # minutes. It needs optimization; for now we pull down the ids
```

Comment:
Ok, interesting. I'm getting slightly off-topic here, but when we [actually resolve](https://github.com/greenriver/hmis-warehouse/blob/fb03878447ac639dbb092ed67ac7f63a9c4833ac/drivers/hmis/app/graphql/types/hmis_schema/project.rb#L118) services on a Project, we ALSO apply a `viewable_by` scope ([here](https://github.com/greenriver/hmis-warehouse/blob/fb03878447ac639dbb092ed67ac7f63a9c4833ac/drivers/hmis/app/graphql/types/hmis_schema/has_services.rb#L37)) which is completely unnecessary. Instead, we could just check once whether the user has access to view services (i.e. enrollment details) in this project. I think with the "HasX" concerns we errred on the side of caution to always include the viewability scope check, at the detriment of performance.

Reply:
I proposed something for that here: https://github.com/greenriver/hmis-warehouse/pull/3680